FoodOrderingSystem.py

```diff
@@ -0,0 +1,700 @@
+    # View revenue of a particular date/month
+
+    def add_food_category(self, category_name, session):
+        """ Insert a new food category """
+        category = FoodCategory(name=category_name)
+        session.add(category)
+        try:
+            session.commit()
+            return category
+        except IntegrityError:
+            session.rollback()
+            raise Exception("Category exist")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def add_food_details(self, category_id, food_name, price, session):
+        """ Insert new food details """
+        details = FoodDetails(category_id=category_id, food_name=food_name, price=price)
+        session.add(details)
+        try:
+            session.commit()
+            return details
+        except IntegrityError:
+           session.rollback()
+           raise Exception("Food exist")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def add_delivery_person(self, delivery_person_name, delivery_person_phone, session):
+        """ Insert new delivery person """
+        details = DeliveryPerson(delivery_person_name=delivery_person_name, delivery_person_phone=delivery_person_phone)
+        session.add(details)
+        try:
+            session.commit()
+            return details
+        except IntegrityError:
+           session.rollback()
+           raise Exception("Delivery person exist")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def assign_deliver_person_to_deliver_order(self, order_id, delivery_person_id, session):
+        """ Update CustOrderStatus table to add deliver person to deliver order """
+        update = session.query(CustOrderStatus).filter(CustOrderStatus.order_id == order_id).update({CustOrderStatus.delivery_person_id: delivery_person_id}, synchronize_session=False)
+        try:
+            session.commit()
+            return update
+        except:
+            session.rollback()
+            raise Exception("Checkout not completed!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def view_orders(self, order_id, session):
+        """ View particular details of an order """
+        try:
+            view = session.query(FoodCategory, FoodDetails, CustOrderSelection, CustomerDetails, CustOrderStatus).\
+                filter(CustomerDetails.cust_id == CustOrderStatus.cust_id).\
+                filter(CustOrderSelection.food_id == FoodDetails.food_id).\
+                filter(FoodCategory.category_id == FoodDetails.category_id).\
+                filter(CustOrderSelection.order_id == CustOrderStatus.order_id).\
+                filter(CustOrderStatus.order_id == order_id)
+                # label('total_price'), func.sum(CustOrderSelection.food_qty*FoodDetails.price)
+                # (func.sum(CustOrderSelection.food_qty*FoodDetails.price).label("total_price"))
```

Comment:
Why you're using filter(...) instead of doing join on the tables in view_orders() and view_order_status() and why you're returning all columns from all tables instead of just the columns that matter?

Reply:
We can use Query.filter() to equate their related columns together. In order to view orders, we need the food category, food name, food quantity, customer name, order id, price per item and total price, therefore we need to join all the relevant tables together. Further details will be found in the main() "elif employee_options == 6:"